mage_ai/frontend/pages/datasets/[slug].tsx

```diff
@@ -37,10 +43,24 @@ function Data() {
     datasetResponse?.statistics,
   ]);
 
+  const suggestionsMemo = useMemo(() => (
+    (datasetResponse?.suggestions || [])
+  ), [
+    datasetResponse?.suggestions,
+  ]);
+
   const [columnHeaderSample, setColumnHeaderSample] = useState([{}]);
   const [rowGroupDataSample, setRowGroupDataSample] = useState({});
   const [metricSample, setMetricSample] = useState({});
   const [statSample, setStatSample] = useState({});
+
+  const [suggestions, setSuggestions] = useState([]);
+
+  // structured as [{ idx, action_data }]
+  const [actions, setActions] = useState([]);
```

Comment:
A type or interface would've been helpful here. Hard to follow the type of data being stored here.

Reply:
Agreed. For the sake of time I didn't end up making interfaces, but we should definitely do that for all types of payloads (eg. for feature sets payload, a single feature set payload, and individual actions/suggestions payload).